src/fileseq/filesequence.py

```diff
@@ -674,6 +682,12 @@ def getPaddingNum(chars):
         try:
             return sum([PAD_MAP[char] for char in chars])
         except KeyError:
+            # Try to establish what the bad character was for the error msg
+            try:
+                for char in chars:
+                    PAD_MAP[char]
+            except KeyError:
+                pass
```

Comment:
What was the case where this would fail to report the last `char` incorrectly? Unless I am reading this wrong, does not not do the same thing as the failed list comprehension?

Reply:
The variable name 'char' only exists within the scope of the list comprehension, so I ..... wait a sec I'm getting a different result when I try this now.

This is odd. I had been getting a NameError when I ran this before, because when I tried to use 'char' in the ValueError message below, 'char' was undefined. Might be tricky to repro that. I'll revert this I think for now.

Reply:
Ok - this code gives different results in python 2 and 3:

```python
PAD_MAP = {"#": 4, "@": 1}
chars = "#@X"

try:
    total = sum([PAD_MAP[char] for char in chars])
    print(char)
except Exception as e:
    print(e)
```

In python2 the 'char' remains in scope outside the list comprehension, whereas in python3, list comprehensions have their own scope.

We could change the 'raise', from/to:
```python
raise ValueError(msg.format(char, str(PAD_MAP.keys())))
raise ValueError(msg.format(chars, str(PAD_MAP.keys())))
```
which is simpler but less helpful.

Reply:
Ah that's a py3 change I didn't know they made. Seems reasonable that they were trying to fix possible scope bugs by previously allowing shadowing.
PAD_MAP is always going to be a tiny data structure. So I think there is no point in trying to preserve a list comp here and it can just for-loop the sum and then have a local char to report in the exception